### DISCUSSION TITLE
MacOS: Connecting to a local .net web api fails.

### DISCUSSION THREAD

**Discussion by Luc1412** [aio-libs/aiohttp#7047] [Q&A]
The following python application works flawless with the same configuration on Windows.

I got a Mac with Python 3.10 setup. I also installed certificates after installing Python, so aiohttp works fine when connecting to external services like Discord.com's API trough the discord.py library.

I build a .net web api, which I want to use in my py app. This .net app issues a TLSv2 SSL cert for localhost. Testing the api trough Postman works fine.

When using this api in my app with a default session I get `ClientConnectorSSLError(ConnectionKey(host='127.0.0.1', port=5067, is_ssl=True, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=None), SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:997)'))`. 
I also tried creating a session with ssl verification disabled: `connector=aiohttp.TCPConnector(verify_ssl=sys.platform != 'darwin')`. This also results in the same error. 

Any thoughts on how to solve that?

**Comment by Dreamsorcerer** [comment]
No idea, but if you look at the traceback, I think you'll see the original exception comes from the `ssl` module. So, maybe widening your search to issues in Python's ssl module might provide more luck in figuring out the problem.

**Reply by webknjaz** [reply]
I'd say set up the TLS context properly to trust that certificate instead of disabling the verification. Could use `trustme` in tests, like us.

**Reply by webknjaz** [reply]
> WRONG_VERSION_NUMBER

This probably means that the TLS version supported by the server is not enabled in the client's TLS context. IIRC there's been some changes in the TLS versions enabled by default in the recent CPython's but one would need to check the changelogs.

**Reply by webknjaz** [reply]
I think this should be solvable by providing own `SSLContext` instance constructed with the vulnerable TLS versions enabled.

**Reply by Dreamsorcerer** [reply]
> I'd say set up the TLS context properly to trust that certificate instead of disabling the verification.

Also worth pointing out that there is no such thing as disabling verification. That option just "relaxes" the verification (i.e. allows self-signed certs) using a `SSLContext` (https://github.com/aio-libs/aiohttp/blob/12f56d7892151154343c6048526d007bbe3a4291/aiohttp/connector.py#L896-L910). So, if something even less restrictive is needed, then passing a `SSLContext` with your exact requirements is necessary.
